openpype/hosts/houdini/plugins/publish/collect_karma_rop.py

```diff
@@ -95,8 +95,9 @@ def replace(match):
             return path
 
         expected_files = []
-        start = instance.data["frameStart"]
-        end = instance.data["frameEnd"]
+        start = instance.data.get("frameStartHandle")
```

Comment:
why .get? do we even consider the case where `frameStartHandle` doesn't exist and this returns `None`?

Reply:
I think `instance.data["frameStartHandle"]` always exists  unless artists set `trange` to render current frame.

Reply:
> I think instance.data["frameStartHandle"] always exists unless artists set trange to render current frame.

I've mentioned it before - even with `render current frame` it makes much more sense to actually COLLECT what frame range that is instead of then suddenly returning no frame range whatsoever.

Current frame should still be `{"frameStart": current_frame, "frameEnd": current_frame}` etc.

Reply:
> Current frame should still be {"frameStart": current_frame, "frameEnd": current_frame} etc.

what about ? 

```python
if node.evalParm("trange") == 0:
    data["frameStartHandle"] = hou.intFrame() 
    data["frameEndHandle"] = hou.intFrame() 
    log.debug(
            "Node '{}' has 'Render current frame' set. "
            "frameStart and frameEnd are set to the current frame".format(node.path())
        )
else: 
    data["frameStartHandle"] = node.evalParm("f1")
    data["frameEndHandle"] = node.evalParm("f2")
    data["byFrameStep"] = node.evalParm("f3")
   
data["handleStart"] = asset_data.get("handleStart", 0)
data["frameStart"] = data["frameStartHandle"] + data["handleStart"]

data["handleEnd"] = asset_data.get("handleEnd", 0)
data["frameEnd"] = data["frameEndHandle"] - data["handleEnd"]

return data
```

Reply:
For the current frame please also add `byFrameStep` just so we produce consistent data keys. So add `data["byFrameStep"] = 1` there.

Reply:
This should also fix these issues 

because frameStart won't fall to zero in validations 
https://github.com/ynput/OpenPype/issues/5488

Bgeo won't complain about absence of frameStart  and frameEnd
https://github.com/ynput/OpenPype/issues/5507

Reply:
yeah my point of the comment was to not use `.get()` if it's not possible that the key doesn't exist so you use `[]` so it raises KeyError if by any reason it doesn't and it's not a more cryptic error later on.

Do we make use of `byFrameStep`? Would be good to use in case you just want to render every other frame for example. This is another of those things that Deadline vanilla implementation allows you for great flexibility and we lose it with OP's submission: https://www.awsthinkbox.com/blog/frame-list-formatting